completion/available/gradle.completion.bash

```diff
@@ -262,22 +269,23 @@ _gradle() {
 			__gradle-set-files-checksum
 
 			# The cache key is md5 sum of all gradle scripts, so it's valid if it exists.
-			if [[ -f $cache_dir/$cache_name.md5 ]]; then
-				local cached_checksum="$(cat $cache_dir/$cache_name.md5)"
+			if [[ -f "$cache_dir/$cache_name.md5" ]]; then
+				local cached_checksum
+				cached_checksum="$(cat "$cache_dir/$cache_name.md5")"
 				local -a cached_tasks
 				if [[ -z $cur ]]; then
-					cached_tasks=($(cat $cache_dir/$cached_checksum))
+					while IFS='' read -r line; do cached_tasks+=("$line"); done < "$cache_dir/$cached_checksum"
 				else
-					cached_tasks=($(grep "^$cur" $cache_dir/$cached_checksum))
+					while IFS='' read -r line; do cached_tasks+=("$line"); done < <(grep "^$cur" "$cache_dir/$cached_checksum")
 				fi
-				COMPREPLY=($(compgen -W "${cached_tasks[*]}" -- "$cur"))
+				while IFS='' read -r line; do COMPREPLY+=("$line"); done < <(compgen -W "${cached_tasks[*]}" -- "$cur")
 			else
 				__gradle-notify-tasks-cache-build
 			fi
 
 			# Regenerate tasks cache in the background
-			if [[ $gradle_files_checksum != "$(cat $cache_dir/$cache_name.md5)" || ! -f $cache_dir/$gradle_files_checksum ]]; then
-				$(__gradle-generate-tasks-cache 1>&2 2> /dev/null &)
+			if [[ $gradle_files_checksum != "$(cat "$cache_dir/$cache_name.md5")" || ! -f "$cache_dir/$gradle_files_checksum" ]]; then
+				"$(__gradle-generate-tasks-cache 1>&2 2> /dev/null &)"
```

Comment:
Since both stdout and stderr are redirected, the command substitution produces an empty string. Then, if you quote it, it tries to run the empty command to result in the `bash: : command not found` error.

Also, I suspect the intent of `1>&2 2>/dev/null` is actually `2> /dev/null 1>&2` because it tries to run the command in the background by suffixing `&`. The meaning of `1>&2 2>/dev/null` is first to connect 1 to external 2 and then connect 2 to `/dev/null`, i.e., stdout of the command still goes to stderr outside the command substitution. Only stderr of the command is connected to `/dev/null`. However, it would be annoying that the background process still outputs something to the terminal. Therefore, it should have been `2>/dev/null 1>&2` (or more compactly, `&>/dev/null`.

Reply:
fixed it in a way that makes more sense.